controllers/ArticleController.js

```diff
@@ -79,31 +79,47 @@ class ArticleController {
   }
 
   /**
-   * get all articles created
+   * get all articles created and use the query
+   * if provided to implement pagination
    * @param {object} req - request object
    * @param {object} res - response object
    * @param {function} next - next function
    * @returns {object} - the found article from database or empty if not found
   */
   static listAllArticles(req, res, next) {
+    const { page } = req;
+    let limit = null;
+    let offset = null;
+
+    if (page) {
+      // calculate offset and limit
+      limit = 10;
```

Comment:
@Lumexralph `limit` should be passed as a parameter but with a default value of 10 if it is null or not passed.

Reply:
thanks for the insight, I will implement the suggestion